invenio_previewer/utils.py

```diff
@@ -29,7 +29,11 @@ def detect_encoding(fp, default=None):
         result = cchardet.detect(sample)
         threshold = current_app.config.get('PREVIEWER_CHARDET_CONFIDENCE', 0.9)
         if result.get('confidence', 0) > threshold:
-            return result.get('encoding', default)
+            encoding = result.get('encoding', default)
+            if encoding == 'ASCII':
+                # if ascii, override to default (usually utf8) in case of unicode beyond checked range
+                encoding = default
+            return encoding
```

Comment:
Had a similar issue. Thanks for that. With https://github.com/inveniosoftware/invenio-previewer/pull/180 , we will have as good rendering as we can.

Because there is a lot of logic, I rewrote it flatter and took care of another weird case where `confidence`/`encoding` of `result` were `None`:

```python
# ...
try:
    chardet_size = current_app.config.get('PREVIEWER_CHARDET_BYTES', 1024)
    threshold = current_app.config.get('PREVIEWER_CHARDET_CONFIDENCE', 0.9)

    sample = fp.read(chardet_size)

    # Result contains 'confidence' and 'encoding'
    result = cchardet.detect(sample)
    confidence = result.get('confidence', 0) or 0
    encoding = result.get('encoding', default) or default

    # if low confidence or ascii, override to default (usually utf8 which is 
    # better in case of unicode beyond checked range)
    if confidence <= threshold or encoding == "ASCII":
        encoding = default

    return encoding
except Exception:
# ...
```

Reply:
That's overall easier to read, and I've switched to that in the PR.